lib/opensusebasetest.pm

```diff
@@ -484,7 +484,8 @@ sub enter_test_text {
 }
 
 sub firewall {
-    return ((is_sle && !sle_version_at_least('15')) || (is_leap && !leap_version_at_least('15.0'))) ? 'SuSEfirewall2' : 'firewalld';
+    my $old_product_versions = (is_sle && !sle_version_at_least('15')) || (is_leap && !leap_version_at_least('15.0'));
+    return ($old_product_versions || get_var('UPGRADE')) ? 'SuSEfirewall2' : 'firewalld';
```

Comment:
For now this will be ok, but as soon as we will start to add an upgrade test for Leap 15 to TW, this won't be true anymore, as Leap 15.0 will be a firewalld based release

Reply:
ah, that was the missing part :-) ok, so for a proper upgrade test there should be a "version_switch" which IIRC we already have for SLE migration tests but not for openSUSE. If we have that then we can have an easier version pointing to the version from which we upgrade and we could check that